setup/air-gapped/infrastructure.md

```diff
@@ -0,0 +1,159 @@
+---
+title: Air-Gapped Network Setup
+description: Learn how to set up a network for air-gapped Coder deployment.
+---
+
+This article will walk you through setting up your network to support an
+air-gapped Coder deployment.
+
+If your network already has the following, you can proceed with your
+installation:
+
+- A certificate authority
+- A domain name service
+- A local Docker Registry
+
+> The code snippets provided in this article are sourced from third-party
+> software packages. While we attempt to keep this article up-to-date, we
+> strongly recommend that you verify the snippets as well before using.
+
+## Creating the local registry and generating a self-signed certificate
+
+You will need to create a local registry to store your Coder images and
+self-signed certificates (you can use self-signed certificates in any
+environment, but they're required for air-gapped deployments).
+
+You can create your local registry to and your self-signed certificate at the
+same time:
+
+```bash
+export REGISTRY_DOMAINNAME=registry.local
+mkdir /certs
+openssl req \
+    -newkey rsa:4096 -nodes -sha256 -keyout /certs/registry.key \
+    -x509 -days 365 -out /certs/registry.crt
+```
+
+The console will prompt you for `Common Name [CN]:`; provide the value that
+matches exactly what you set with your DNS. For the volume mounted at
+`/var/lib/registry`, make sure that it has at least 10 GB for Coder images.
+
+Start the registry container that you just created:
+
+```bash
+docker run -d -p 443:5000 \
+    -e REGISTRY_HTTP_TLS_CERTIFICATE=/certs/registry.crt \
+    -e REGISTRY_HTTP_TLS_KEY=/certs/registry.key \
+    -v /certs:/certs \
+    -v /var/lib/docker/registry:/var/lib/registry \
+    registry:2
+```
+
+> For the volume mounted at `/var/lib/registry` make sure it can store 10+ GB
+> for just Coder images.
+
+## Configuring the Kubernetes Node
+
+Before the Kubernetes node can accept your certificate, you'll need to mark your
+`registry.crt` file as trusted. The specific locations where you need to have
+this file depends on your Linux distribution and the container runtime:
+
```

Comment:
I thought this section was a bit confusing; can you expand a bit more on what the end-user needs to consider so the cert's trusted?

Reply:
I'm not experienced with this piece just yet, so I will defer to @mterhar.

Reply:
The multiple locations in the file below are all the places that I needed to set for an ubuntu base image to be okay with the self-signed cert. Maybe just referring to that list as a partial set that may require additions depending on the OS and kubernetes distribution.